kernel/src/transaction/mod.rs

```diff
@@ -850,14 +864,21 @@ impl Transaction {
 pub struct WriteContext {
     target_dir: Url,
     schema: SchemaRef,
```

Comment:
let's rename this (and associated functions) `logical_schema` to make things more clear